### PR TITLE
Yifei's homework

### PR DESCRIPTION
**Pull request by leftan**
Thanks in advance for the review and feedback, Daniel.

### CODE REVIEW

File: D04_ex00.py [hunk 1]
```diff
@@ -11,19 +11,44 @@
...
+            print("Your number is not in the range. Try again!")
+        elif input_num  == answer:
+            print('You won! The guess number is ' + str(answer)) 
+            break
+        elif input_num > answer:
+            print('Your number is too high. Try again!')
+            count += 1
+            print('you can try ' + str(5 - count) + ' times.')
+            continue
+        elif input_num < answer:
```
Comment (danielsgriffin) [inline D04_ex00.py]: else

File: D04_ex00.py [hunk 2]
```diff
@@ -11,19 +11,44 @@
...
+            print('Your number is too low. Try again!')
+            count += 1
+            print('you can try ' + str(5 - count) + ' times.')
+            continue
```
Comment (danielsgriffin) [inline D04_ex00.py]: Is this necessary?

File: HW04_ch08_ex00.py [hunk 3]
```diff
@@ -18,6 +18,14 @@
 ###############################################################################
 # Imports
 
+def count(word, letter):
+    '''word, letter: str'''
```
Comment (danielsgriffin) [inline HW04_ch08_ex00.py]: nice

File: HW04_ch08_ex04.py [hunk 4]
```diff
@@ -34,25 +34,24 @@ def any_lowercase2(s):
 
 
 def any_lowercase3(s):
-    """Explain what is wrong, if anything, here.
+    """variable 'flag' doesn't have an initial value.
```
Comment (danielsgriffin) [inline HW04_ch08_ex04.py]: that is not *the* problem

Comment (danielsgriffin) [inline HW04_ch08_ex04.py]: there is another problem.

File: HW04_ch08_ex05.py [hunk 5]
```diff
@@ -2,3 +2,41 @@
 # See Chapter 8: Strings Exercise 5 for the task.
 # Please do provide function calls that test/demonstrate your
 # function.
+
+def shift_letter(letter, num):
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]: Good job.

**Comment by danielsgriffin** [comment]
Good job.

File: HW04_string2.py [hunk 6]
```diff
@@ -42,8 +51,7 @@ def not_bad(s):
 # Given 2 strings, a and b, return a string of the form
 #  a-front + b-front + a-back + b-back
 def front_back(a, b):
-    # +++your code here+++
-    return
+    return a[:(len(a) - len(a)//2)] + b[:(len(b) - len(b)//2)] + a[(len(a) - len(a)//2):] + b[(len(b) - len(b)//2):]
```
Comment (feladie) [inline HW04_string2.py]: I like how succinct this code is!

**Pull request closed without merging**